commit 1ad303f84ff37e145e4cc727538d86bf877ab8b9
Author: Jean-Luc Mackail <fuzzyquils@scrapyard.link>
Date:   Thu Jan 11 16:46:35 2024 +1000

    proton: Add support for Winesync via PROTON_USE_WINESYNC.
    
    This should allow esync/fsync to be the default while
    allowing testing of fastsync patches with the environment
    variable PROTON_USE_WINESYNC set to 1.

diff --git a/proton b/proton
index f93f545f..2e0199f0 100755
--- a/proton
+++ b/proton
@@ -1303,6 +1303,10 @@ class Session:
         if not "noforcelgadd" in self.compat_config:
             self.compat_config.add("forcelgadd")
 
+        #turn off fastsync by default unless it is enabled via env
+        if not "winesync" in self.compat_config:
+            self.compat_config.add("nowinesync")
+
     def init_wine(self):
         if "HOST_LC_ALL" in self.env and len(self.env["HOST_LC_ALL"]) > 0:
             #steam sets LC_ALL=C to help some games, but Wine requires the real value
@@ -1425,6 +1429,7 @@ class Session:
         self.check_environment("PROTON_NO_D3D10", "nod3d10")
         self.check_environment("PROTON_NO_ESYNC", "noesync")
         self.check_environment("PROTON_NO_FSYNC", "nofsync")
+        self.check_environment("PROTON_USE_WINESYNC", "winesync")
         self.check_environment("PROTON_FORCE_LARGE_ADDRESS_AWARE", "forcelgadd")
         self.check_environment("PROTON_OLD_GL_STRING", "oldglstr")
         self.check_environment("PROTON_NO_WRITE_WATCH", "nowritewatch")
@@ -1451,6 +1456,12 @@ class Session:
         else:
             self.env["WINEFSYNC"] = "1"
 
+        if "winesync" in self.compat_config:
+            self.env.pop("WINE_DISABLE_FAST_SYNC", "")
+        else:
+            if "nowinesync" in self.compat_config:
+                self.env["WINE_DISABLE_FAST_SYNC"] = "1"
+
         if "nowritewatch" in self.compat_config:
             self.env["WINE_DISABLE_WRITE_WATCH"] = "1"
 
